src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/util/BigQuerySQLEngineUtils.java

```diff
@@ -292,4 +296,79 @@ public static boolean isValidIdentifier(String identifier) {
     labels.put("pushdown_operation", operation);
     return Collections.unmodifiableMap(labels);
   }
+
+  /**
+   * Logs information about a BigQUery Job execution using a specified Logger instance
+   *
+   * @param job BigQuery Job
+   */
+  public static void logJobMetrics(Job job) {
+    // Ensure job has statistics information
+    if (job.getStatistics() == null) {
+      LOG.warn("No statistics were found for BigQuery job {}", job.getJobId());
+    }
+
+    String startTimeStr = getISODateTimeString(job.getStatistics().getStartTime());
+    String endTimeStr = getISODateTimeString(job.getStatistics().getEndTime());
+    String executionTimeStr = getExecutionTimeString(job.getStatistics().getStartTime(),
+                                                     job.getStatistics().getEndTime());
+
+    // Print detailed query statistics if available
+    if (job.getStatistics() instanceof JobStatistics.QueryStatistics) {
+      JobStatistics.QueryStatistics queryStatistics = (JobStatistics.QueryStatistics) job.getStatistics();
+      LOG.info("Metrics for job {}:\n" +
+                 " Start: {} ,\n" +
+                 " End: {} ,\n" +
+                 " Execution time: {} ,\n" +
+                 " Processed Bytes: {} ,\n" +
```

Comment:
Could you also log `numDmlAffectedRows`? I think it should be more or less corresponding to output rows and we mostly talk about rows in CDAP